src/Reducers/cartReducer.js

```diff
@@ -0,0 +1,148 @@
+export const cartReducer = (state, action) => {
+  switch (action.type) {
+    case "ADD_TO_CART":
+      return {
+        ...state,
+        cartItems: state.cartItems.concat({ ...action.payload }),
+        productData: state.productData.map((item) =>
```

Comment:
operations on productData is sort of redundant. You're already using functions for toggle actions so you can remove this . It will minimize the your code quite a lot.